Duplicate @Test annotation added to method

I have created a simple recipe to add a Junit5 @Test annotation to a method that has a legacy @testName custom javadoc tag using the latest 7.40.8 openrewrite release. A simple `RewriteTest` shows the annotation being added twice. In debugging this, the reason is that the `AnnotationMatcher("@org.junit.jupiter.api.Test")` used to validate that a method does not already have an annotation fails to match the existing annotation because the `matchesAnnotationName` call to `TypeUtils.asFullyQualified(annotation.getType())` returns null and the `annotation.getType()` returns Unknown.

Full test repo is here:
https://github.com/starksm64/tck-rewrite-tools.git

Output of mvn test:
```bash
[INFO] Running tck.conversion.rewrite.ConvertTestNameTest
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Visting(someTestMethod)
Found unknown block tag: testName:
Added @Test annotation?
MethodDeclaration{test.somepkg.SomeTestClass{name=someTestMethod,return=void,parameters=[]}}
Visting(someTestMethod)
Found unknown block tag: testName:
Added @Test annotation?
MethodDeclaration{test.somepkg.SomeTestClass{name=someTestMethod,return=void,parameters=[]}}
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.735 s <<< FAILURE! - in tck.conversion.rewrite.ConvertTestNameTest
[ERROR] tck.conversion.rewrite.ConvertTestNameTest.addTestAnnotation  Time elapsed: 0.729 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: 
[Expected recipe to complete in 1 cycle, but took at least one more cycle. Between the last two executed cycles there were changes to "test/somepkg/SomeTestClass.java"] 
expected: 
  "package test.somepkg;
  
  
  public class SomeTestClass {
  
      /**
      @testName: someTestMethod
      */
      @Test
      public void someTestMethod() {
      }
  }"
 but was: 
  "package test.somepkg;
  
  
  public class SomeTestClass {
  
      /**
      @testName: someTestMethod
      */
      @Test
      @Test
      public void someTestMethod() {
      }
  }"
        at org.openrewrite.test.RecipeSchedulerCheckingExpectedCycles.scheduleVisit(RecipeSchedulerCheckingExpectedCycles.java:69)
        at org.openrewrite.RecipeScheduler.scheduleRun(RecipeScheduler.java:106)
        at org.openrewrite.Recipe.run(Recipe.java:498)
        at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:320)
        at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:131)
        at org.openrewrite.test.RewriteTest.rewriteRun(RewriteTest.java:126)
        at tck.conversion.rewrite.ConvertTestNameTest.addTestAnnotation(ConvertTestNameTest.java:18)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ConvertTestNameTest.addTestAnnotation:18 [Expected recipe to complete in 1 cycle, but took at least one more cycle. Between the last two executed cycles there were changes to "test/somepkg/SomeTestClass.java"] 
expected: 
  "package test.somepkg;
  
  
  public class SomeTestClass {
  
      /**
      @testName: someTestMethod
      */
      @Test
      public void someTestMethod() {
      }
  }"
 but was: 
  "package test.somepkg;
  
  
  public class SomeTestClass {
  
      /**
      @testName: someTestMethod
      */
      @Test
      @Test
      public void someTestMethod() {
      }
  }"
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
```

Code for the JavaIsoVisitor that adds the annotation [ConvertJavaTestNameVisitor.java](https://github.com/starksm64/tck-rewrite-tools/blob/02c3b59867163c3bebc212f8eaa83ea871c19939/src/main/java/tck/jakarta/platform/rewrite/ConvertJavaTestNameVisitor.java):
```java
package tck.jakarta.platform.rewrite;

import org.openrewrite.java.AnnotationMatcher;
import org.openrewrite.java.JavaIsoVisitor;
import org.openrewrite.java.JavaTemplate;
import org.openrewrite.java.tree.Comment;
import org.openrewrite.java.tree.J;
import org.openrewrite.java.tree.Javadoc;

import java.util.Comparator;

/**
 * Find methods marked with a @testName javadoc comment and add a Junit5 @Test annotation
 * @param <ExecutionContext>
 */
public class ConvertJavaTestNameVisitor<ExecutionContext> extends JavaIsoVisitor<ExecutionContext> {
    private final AnnotationMatcher TEST_ANN_MATCH = new AnnotationMatcher("@org.junit.jupiter.api.Test");

    private final JavaTemplate testAnnotationTemplate =
            JavaTemplate.builder(this::getCursor, "@Test")
                    .imports("org.junit.jupiter.api.Test")
                    .build();


    public ConvertJavaTestNameVisitor() {
    }

    @Override
    public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ec) {
        String methodName = method.getSimpleName();
        if(method.getAllAnnotations().stream().anyMatch(TEST_ANN_MATCH::matches)) {
            System.out.printf("Visting(%s) skipped due to @Test\n", methodName);
            return super.visitMethodDeclaration(method, ec);
        }
        System.out.printf("Visting(%s)\n", methodName);
        for(Comment c : method.getComments()) {
            if(c instanceof Javadoc.DocComment) {
                for(Javadoc jd : ((Javadoc.DocComment) c).getBody()) {
                    if(jd instanceof Javadoc.UnknownBlock) {
                        String name = ((Javadoc.UnknownBlock) jd).getName();
                        System.out.printf("Found unknown block tag: %s\n", name);
                        if(name.equals("testName:")) {
                            method = method.withTemplate(testAnnotationTemplate,
                                    method.getCoordinates().addAnnotation(Comparator.comparing(J.Annotation::getSimpleName)));
                            System.out.println("Added @Test annotation?\n"+method);
                        }
                    }
                }
            }
        }

        return super.visitMethodDeclaration(method, ec);
    }

}
```

--- a/src/main/java/tck/jakarta/platform/rewrite/ConvertJavaTestNameVisitor.java
+++ b/src/main/java/tck/jakarta/platform/rewrite/ConvertJavaTestNameVisitor.java
@@ -2,6 +2,7 @@ package tck.jakarta.platform.rewrite;
 
 import org.openrewrite.java.AnnotationMatcher;
 import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
 import org.openrewrite.java.JavaTemplate;
 import org.openrewrite.java.tree.Comment;
 import org.openrewrite.java.tree.J;
@@ -18,12 +19,9 @@ public class ConvertJavaTestNameVisitor<ExecutionContext> extends JavaIsoVisitor
 
     private final JavaTemplate testAnnotationTemplate =
             JavaTemplate.builder(this::getCursor, "@Test")
+                    .javaParser(JavaParser.fromJavaVersion().classpath("junit-jupiter-api"))
                     .imports("org.junit.jupiter.api.Test")
                     .build();
-
-
-    public ConvertJavaTestNameVisitor() {
-    }
 
     @Override
     public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ec) {
@@ -38,11 +36,13 @@ public class ConvertJavaTestNameVisitor<ExecutionContext> extends JavaIsoVisitor
                 for(Javadoc jd : ((Javadoc.DocComment) c).getBody()) {
                     if(jd instanceof Javadoc.UnknownBlock) {
                         String name = ((Javadoc.UnknownBlock) jd).getName();
-                        System.out.printf("Found unknown block tag: %s\n", name);
                         if(name.equals("testName:")) {
                             method = method.withTemplate(testAnnotationTemplate,
                                     method.getCoordinates().addAnnotation(Comparator.comparing(J.Annotation::getSimpleName)));
+                            maybeAddImport("org.junit.jupiter.api.Test");
                             System.out.println("Added @Test annotation?\n"+method);
+                        } else {
+                            System.out.printf("Unknown block tag: %s\n", name);
                         }
                     }
                 }

--- a/src/test/java/tck/conversion/rewrite/ConvertTestNameTest.java
+++ b/src/test/java/tck/conversion/rewrite/ConvertTestNameTest.java
@@ -1,16 +1,19 @@
 package tck.conversion.rewrite;
 
 import org.junit.jupiter.api.Test;
+import org.openrewrite.java.JavaParser;
 import org.openrewrite.test.RecipeSpec;
 import org.openrewrite.test.RewriteTest;
 import tck.jakarta.platform.rewrite.ConvertJavaTestNameRecipe;
 
 import static org.openrewrite.java.Assertions.java;
 
-public class ConvertTestNameTest implements RewriteTest {
+class ConvertTestNameTest implements RewriteTest {
     @Override
     public void defaults(RecipeSpec spec) {
-        spec.recipe(new ConvertJavaTestNameRecipe());
+        spec
+                .recipe(new ConvertJavaTestNameRecipe())
+                .parser(JavaParser.fromJavaVersion().classpath("junit-jupiter-api"));
     }
 
     @Test
@@ -18,33 +21,32 @@ public class ConvertTestNameTest implements RewriteTest {
         rewriteRun(
                 java(
                         """
-                            package test.somepkg;
-        
-        
-                            public class SomeTestClass {
-  
-                                /**
-                                @testName: someTestMethod
-                                */
-                                public void someTestMethod() {
-                                }
-                            }
-                        """,
+                                    package test.somepkg;
+                                    
+                                    public class SomeTestClass {
+                                  
+                                        /**
+                                        @testName: someTestMethod
+                                        */
+                                        public void someTestMethod() {
+                                        }
+                                    }
+                                """,
                         """
-                            package test.somepkg;
-                            
-                            import org.junit.jupiter.api.Test;
-        
-                            public class SomeTestClass {
-  
-                                /**
-                                @testName: someTestMethod
-                                */
-                                @Test
-                                public void someTestMethod() {
-                                }
-                            }
-                        """
+                                    package test.somepkg;
+                                    
+                                    import org.junit.jupiter.api.Test;
+                                    
+                                    public class SomeTestClass {
+                                  
+                                        /**
+                                        @testName: someTestMethod
+                                        */
+                                        @Test
+                                        public void someTestMethod() {
+                                        }
+                                    }
+                                """
                 )
         );
     }

